SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialPlayerSpawner.cpp

```diff
@@ -198,40 +198,58 @@ void USpatialPlayerSpawner::ReceivePlayerSpawnRequestOnServer(const Worker_Comma
 
 void USpatialPlayerSpawner::FindPlayerStartAndProcessPlayerSpawn(Schema_Object* SpawnPlayerRequest, const PhysicalWorkerName& ClientWorkerId)
 {
-	// If load-balancing is enabled AND the strategy dictates that another worker should have authority over
-	// the chosen PlayerStart THEN the spawn request is forwarded to that worker to prevent an initial player
-	// migration. Immediate player migrations can still happen if
-	// 1) the load-balancing strategy has different rules for PlayerStart Actors and Characters / Controllers /
-	// Player States or,
-	// 2) the load-balancing strategy can change the authoritative virtual worker ID for a PlayerStart Actor
-	// during the lifetime of a deployment.
-	if (NetDriver->LoadBalanceStrategy != nullptr)
+	// If the load balancing strategy dictates that this worker should have authority over the chosen PlayerStart THEN the spawn is handled locally,
+	// Else if the the PlayerStart is handled by another worker THEN forward the request to that worker to prevent an initial player migration,
+	// Else if a PlayerStart can't be found THEN we could be on the wrong worker type, so forward to the GameMode authoritative server.
+	// 
+	// This implementation depends on:
+	// 1) the load-balancing strategy having the same rules for PlayerStart Actors and Characters / Controllers / Player States or,
+	// 2) the authoritative virtual worker ID for a PlayerStart Actor not changing during the lifetime of a deployment.
+	check (NetDriver->LoadBalanceStrategy != nullptr)
+
+	// We need to specifically extract the URL from the PlayerSpawn request for finding a PlayerStart.
+	const FURL Url = PlayerSpawner::ExtractUrlFromPlayerSpawnParams(SpawnPlayerRequest);
+
+	// Find a PlayerStart Actor on this server.
+	AActor* PlayerStartActor = NetDriver->GetWorld()->GetAuthGameMode()->FindPlayerStart(nullptr, Url.Portal);
```

Comment:
Should we add some documentation to FindPlayerStart_Implementation (and blueprint functions K2_FindPlayerStart and ChoosePlayerStart) in our engine fork that we pass nullptr here for load balancing purposes?

Reply:
Yea this sounds useful

Reply:
Going to add to zoning docs for now instead. Adding comments alone to engine branches sounds like a lot of effort as something that might also change 😅

Reply:
Added